core-web/libs/sdk/experiments/src/lib/parser/parser.ts

```diff
@@ -63,3 +63,122 @@ const verifyRegex = (regexToCheck: string | null, href: string): boolean => {
+ *
+ * @returns { AssignedExperiments } - The parsed experiment data ready for storing.
+ *
+ * Following cases are handled -
+ * 1) When new Data is received without Old data. This is assumed to be the first time data is received.
+ * 2) When only old data is received, implying that the timestamp hasn't expired and the data is available in IndexedDB.
+ *    The data is verified and any expired experiment is removed before being assigned to dataToStorage.
+ * 3) When both old data and new data is present. This implies that the record existed in IndexedDB but was fetched because the flag had expired.
+ *    Additional operations are performed to add expiry time to experiments, merging all experiments, and storing them.
+ *
+ * There could be scenarios where none of these conditions are met, in that case, dataToStorage will be the default empty object.
+ */
+export const parseData = (
+    fetchExperiments: AssignedExperiments | null,
+    storedExperiments: AssignedExperiments | null
+): AssignedExperiments => {
+    let dataToStorage: AssignedExperiments = {} as AssignedExperiments;
+
+    if (fetchExperiments && !storedExperiments) {
+        // TODO: Use fetchExperiment instead fetchExperimentsNoNoneExperimentID when the endpoint dont retrieve NONE experiment
+        // https://github.com/dotCMS/core/issues/27905
+        const fetchExperimentsNoNoneExperimentID: AssignedExperiments = {
+            ...fetchExperiments,
+            experiments: fetchExperiments?.experiments
+                ? fetchExperiments.experiments.filter((experiment) => experiment.id !== 'NONE')
+                : []
+        };
+
+        dataToStorage = {
+            ...fetchExperimentsNoNoneExperimentID,
+            experiments: [
+                ...setExpireTimeNewExperiments(fetchExperimentsNoNoneExperimentID.experiments)
+            ]
+        };
+    }
+
+    if (!fetchExperiments && storedExperiments) {
+        dataToStorage = {
+            ...storedExperiments,
+            experiments: getUnexpiredExperiments(storedExperiments.experiments)
+        };
+    }
+
+    if (fetchExperiments && storedExperiments) {
+        // TODO: Use fetchExperiment instead fetchExperimentsNoNoneExperimentID when the endpoint dont retrieve NONE experiment
+        // https://github.com/dotCMS/core/issues/27905
+        const fetchExperimentsNoNoneExperimentID: AssignedExperiments = {
+            ...fetchExperiments,
+            experiments: fetchExperiments?.experiments
+                ? fetchExperiments.experiments.filter((experiment) => experiment.id !== 'NONE')
+                : []
+        };
+        dataToStorage = {
+            ...fetchExperimentsNoNoneExperimentID,
+            experiments: [
+                ...setExpireTimeNewExperiments(fetchExperimentsNoNoneExperimentID.experiments),
+                ...getUnexpiredExperiments(storedExperiments.experiments)
+            ]
+        };
+    }
```

Comment:
the first and the third `if` seems to be duplicated logic. Maybe you can check for `fetchExperiments` and internally add the `getUnexpiredExperiments` if the `storedExperiments` exists.

This way you will have just 2 if statements

Reply:
Yes, you are right, but I made the 3 ifs, so that it can be explicitly seen what I am doing for, so the cognitive complexity is less than having 2 nested ifs.